plugins/node/opentelemetry-instrumentation-bunyan/src/OpenTelemetryBunyanStream.ts

```diff
@@ -0,0 +1,150 @@
+  info: INFO,
+  warn: WARN,
+  error: ERROR,
+  fatal: FATAL,
+};
+const nameFromLevel: { [level: number]: string } = {};
+Object.keys(levelFromName).forEach(function (name) {
+  nameFromLevel[levelFromName[name as LogLevelString]] = name;
+});
+
+const OTEL_SEV_NUM_FROM_BUNYAN_LEVEL: { [level: number]: SeverityNumber } = {
+  [TRACE]: SeverityNumber.TRACE,
+  [DEBUG]: SeverityNumber.DEBUG,
+  [INFO]: SeverityNumber.INFO,
+  [WARN]: SeverityNumber.WARN,
+  [ERROR]: SeverityNumber.ERROR,
+  [FATAL]: SeverityNumber.FATAL,
+};
+
+const EXTRA_SEV_NUMS = [
+  SeverityNumber.TRACE2,
+  SeverityNumber.TRACE3,
+  SeverityNumber.TRACE4,
+  SeverityNumber.DEBUG2,
+  SeverityNumber.DEBUG3,
+  SeverityNumber.DEBUG4,
+  SeverityNumber.INFO2,
+  SeverityNumber.INFO3,
+  SeverityNumber.INFO4,
+  SeverityNumber.WARN2,
+  SeverityNumber.WARN3,
+  SeverityNumber.WARN4,
+  SeverityNumber.ERROR2,
+  SeverityNumber.ERROR3,
+  SeverityNumber.ERROR4,
+  SeverityNumber.FATAL2,
+  SeverityNumber.FATAL3,
+  SeverityNumber.FATAL4,
+];
+
+function severityNumberFromBunyanLevel(lvl: number) {
+  // Fast common case: one of the known levels
+  const sev = OTEL_SEV_NUM_FROM_BUNYAN_LEVEL[lvl];
+  if (sev !== undefined) {
+    return sev;
+  }
+
+  // Otherwise, scale the Bunyan level range -- 10 (TRACE) to 70 (FATAL+10)
+  // -- onto the extra OTel severity numbers (TRACE2, TRACE3, ..., FATAL4).
+  // Values below bunyan.TRACE map to SeverityNumber.TRACE2, which may be
+  // considered a bit weird, but it means the unnumbered levels are always
+  // just for exactly values.
+  const relativeLevelWeight = (lvl - 10) / (70 - 10);
+  const otelSevIdx = Math.floor(relativeLevelWeight * EXTRA_SEV_NUMS.length);
+  const cappedOTelIdx = Math.min(
+    EXTRA_SEV_NUMS.length - 1,
+    Math.max(0, otelSevIdx)
+  );
+  const otelSevValue = EXTRA_SEV_NUMS[cappedOTelIdx];
+  return otelSevValue;
+}
+
+/**
+ * A Bunyan stream for sending log records to the OpenTelemetry Logs Bridge API.
+ */
+export class OpenTelemetryBunyanStream {
+  private _otelLogger: Logger;
+
+  constructor(name?: string, version?: string, options?: LoggerOptions) {
+    this._otelLogger = logs.getLogger(
+      name || DEFAULT_INSTRUMENTATION_SCOPE_NAME,
+      version || DEFAULT_INSTRUMENTATION_SCOPE_VERSION,
+      options
+    );
+  }
+
+  /**
+   * Convert from https://github.com/trentm/node-bunyan#log-record-fields
+   * to https://opentelemetry.io/docs/specs/otel/logs/data-model/
+   *
+   * Dev Notes:
+   * - We drop the Bunyan 'v' field. It is meant to indicate the format
+   *   of the Bunyan log record. FWIW, it has always been `0`.
+   * - Some Bunyan fields would naturally map to Resource attributes:
+   *      hostname -> host.name
+   *      name -> service.name
+   *      pid -> process.pid
+   *   However, AFAIK there isn't a way to influence the LoggerProvider's
+   *   `resource` from here.
+   * - Strip the `trace_id` et al fields that may have been added by the
+   *   the _emit wrapper.
+   */
```

Comment:
> What's your opinion on dropping `hostname` and `pid`?

That's a great idea. I've done so in commit 3638f72.

Because `HostDetector` is not used by default by the NodeSDK, I've added to the "examples/telemetry.js" how to explicitly use it to get a `host.name` resource attribute for the elided Bunyan `hostname` field.

# side questions

(Also let me know if these are better asked in Slack, separate issues, or in the SIG meeting.)

1. Do you happen to know why `HostDetector` is not part of the default set of NodeSDK detectors?
2. Is it a bug that the `SimpleLogRecordProcessor` does not wait for pending async resources, as the `SimpleSpanProcessor` does?

Reply:
> > What's your opinion on dropping `hostname` and `pid`?
> 
> That's a great idea. I've done so in commit [3638f72](https://github.com/open-telemetry/opentelemetry-js-contrib/commit/3638f7212b901b235c3445e78334bc433e9e1671).
> 
> Because `HostDetector` is not used by default by the NodeSDK, I've added to the "examples/telemetry.js" how to explicitly use it to get a `host.name` resource attribute for the elided Bunyan `hostname` field.
> # side questions
> 
> (Also let me know if these are better asked in Slack, separate issues, or in the SIG meeting.)

I like having these question start out on issues and PRs over Slack and SIG meetings as it's easier to reference it the future, but more importantly other approvers or maintainers can answer on my behalf in case I'm unavailable (due to time-zone differences for instance, as many contributors tend to be US-based but I'm Europe). 🙂 Then we can also bring it up in SIG meetings as necessary.

> 1. Do you happen to know why `HostDetector` is not part of the default set of NodeSDK detectors?

There's no specific reason AFAIK. I think it was just not added to NodeSDK it when it was initially implemented in https://github.com/open-telemetry/opentelemetry-js/pull/2921. I'll create an issue in the core repo to track this.

> 2. Is it a bug that the `SimpleLogRecordProcessor` does not wait for pending async resources, as the `SimpleSpanProcessor` does?

Yes it is! That's a great catch. Just checked and the `BatchLogRecordProcessor` also does not do it, but it should. I'll also create an issue.

Reply:
1. https://github.com/open-telemetry/opentelemetry-js/issues/4281
2. https://github.com/open-telemetry/opentelemetry-js/issues/4282

Reply:
Thanks for opening the issues!